suites/apis/clientCredGrantTest.js

```diff
@@ -0,0 +1,148 @@
+/* eslint-disable max-len */
+
+/*
+
+1. create an OIDC client with the client_credential grant
+fence-create with --grant-types `client_credentials`
+which return client_id and secret_id
+
+2. run usersync (The client will have been previously added to the Jenkins user.yaml and granted access to
+what it will need for the test, but i think usersync will not grant the access if the client does not
+exist in the Fence DB. So you need to run usersync again during the test, after creating the client)
+
+3. get a client access token using the client ID and secret.
+
+4. hit Fence’s /user endpoint and check the access
+
+5. create an access request in Requestor
+
+6. list Requestor requests
+
+7. update an access request in Requestor
+
+8. AfterSuite - delete the create client in step 1 with `client-delete` command
+also delete the requests created in requestor
+*/
+
+Feature('Client_Credentials Grant Type @requires-fence');
+
+const { expect } = require('chai');
+const { runUserSync, checkPod, getAccessTokenHeader } = require('../../utils/apiUtil.js');
+const { Bash, takeLastLine } = require('../../utils/bash.js');
+const requestorTasks = require('../../services/apis/requestor/requestorTasks.js');
+const requestorProps = require('../../services/apis/requestor/requestorProps.js');
+
+const bash = new Bash();
+
+const clientName = 'jenkinsClientTester';
+
+async function getAccessToken(clientID, secretID) {
+  const tokenReq = bash.runCommand(`curl --user "${clientID}:${secretID}" --request POST "https://${process.env.HOSTNAME}/user/oauth2/token?grant_type=client_credentials" -d scope="openid user"`);
+  if (process.env.DEBUG === 'true') {
+    console.log(`data : ${tokenReq}`);
+  }
+  const tokens = JSON.parse(tokenReq);
+  const accessToken = tokens.access_token;
+  console.log(`###Access_token : ${accessToken}`);
+  if (process.env.DEBUG === 'true') {
+    console.log(`###Access_Token : ${accessToken}`);
+  }
+  return accessToken;
+}
+
+function deleteClient() {
```

Comment:
there's already a `deleteClient` function in `fenceProps` - could we move it to `fenceTasks` and reuse it here?